client/js/options.js

```diff
@@ -9,22 +9,29 @@ const tz = require("./libs/handlebars/tz");
 const windows = $("#windows");
 const chat = $("#chat");
 
-const options = $.extend({
+const defaultOptions = {
+	autocomplete: true,
 	coloredNicks: true,
-	condense: true,
 	desktopNotifications: false,
+	highlights: [],
 	links: true,
 	motd: true,
 	notification: true,
 	notifyAllMessages: false,
-	optional: true,
 	showSeconds: false,
+	statusMessages: "shown",
 	theme: $("#theme").attr("href").replace(/^themes\/(.*).css$/, "$1"), // Extracts default theme name, set on the server configuration
 	thumbnails: true,
 	userStyles: userStyles.text(),
-	highlights: [],
-	autocomplete: true
-}, JSON.parse(storage.get("settings")));
+};
+const userOptions = JSON.parse(storage.get("settings"));
+const options = {};
+
+for (const key in defaultOptions) {
+	if (userOptions[key]) {
+		options[key] = userOptions[key];
+	}
+}
```

Comment:
I noticed that my settings page was broken, and it wasn't code: I had an old options set in local storage, and it was being carried over, then picked up by the UI. This ensures that only the options defined in the defaults can be overridden by what's in local storage.
This will be useful when a given options takes the same name as an old one by accident. Maybe that was my case, not sure, but I figured a proper fix would be nicer than just wipe local storage.